docstore/mongodocstore/urls.go

```diff
@@ -34,27 +33,25 @@ func init() {
 // defaultDialer dials a default Mongo server based on the environment variable
 // MONGO_SERVER_URL.
 type defaultDialer struct {
-	init   sync.Once
-	opener *URLOpener
-	err    error
+	mongoServerURL string
+	opener         *URLOpener
+	err            error
 }
 
 func (o *defaultDialer) OpenCollectionURL(ctx context.Context, u *url.URL) (*docstore.Collection, error) {
-	o.init.Do(func() {
-		serverURL := os.Getenv("MONGO_SERVER_URL")
+	serverURL := os.Getenv("MONGO_SERVER_URL")
```

Comment:
This is not thread-safe (OpenCollectionURL can be called concurrently on the same `o`).

You can protect it with a sync.Mutex.

Reply:
makes sense, updated 👍🏼